pydantic/_internal/_model_construction.py

```diff
@@ -210,6 +224,8 @@ def __getattr__(self, item: str) -> Any:
             private_attributes = self.__dict__.get('__private_attributes__')
             if private_attributes and item in private_attributes:
                 return private_attributes[item]
+            if item == '__pydantic_deprecated_fields__':
+                return self.__dict__.get('__pydantic_deprecated_fields__')
```

Comment:
Why is this necessary? It feels to me like we should just always set this on the class so that it can be accessed without hitting `__getattr__`. We do special handling of `__pydantic_core_schema__` below because of challenges with needing to defer schema building due to recursive references, etc., but it seems to me that `__pydantic_deprecated_fields__` shouldn't require special treatment like this. Maybe I'm overlooking something?

Reply:
I see it's because of the `__getattribute__` accessing the `__pydantic_deprecated_fields__` attribute. Hopefully as part of dealing with the performance consequences of using `__getattribute__` mentioned below we can drop the need for this.

Reply:
Yes there is some context in this comment: https://github.com/pydantic/pydantic/pull/8237#issuecomment-1871441003, it was indeed tricky to handle as I had recursion errors and with `__getattribute__` the implementation isn't that pretty.